modules/javafx.web/src/ios/java/javafx/scene/web/JSONDecoder.java

```diff
@@ -98,9 +98,9 @@ private Object decode(CharacterIterator it) {
         } else {
             long val = Long.parseLong(sNum);
             if ((val <= Integer.MAX_VALUE) && (Integer.MIN_VALUE <= val)) {
-                return new Integer((int) val);
+                return Integer.valueOf(int) val);
```

Comment:
Autobox?

Reply:
I think, when LHS is of type `Object`, it is good to be explicit on creating Object, it improves readability.
In this case, type of val is `long`, so autobox would create an object of `Long` not `Integer`.

Reply:
RIght, this can lead to bugs if you aren't careful. See the discussion in [JDK-8154213](https://bugs.openjdk.java.net/browse/JDK-8154213). I recommend autoboxing only for those cases where the type is clearly specified.